src/gradualizer_db.erl

```diff
@@ -608,7 +608,9 @@ get_beam_map() ->
                 {match, [Mod]} ->
                     {true, {list_to_atom(Mod), Filename}};
                 nomatch ->
-                    false
+                    false;
+                _ ->
+                    erlang:error({unreachable, "check re:run/3 opts above - this should not happen"})
```

Comment:
Debugging code? We already get a case clause error without this, so I don't think we should add this explicit code. Or is this just to silence a gradualizer error?

Reply:
> is this just to silence a gradualizer error?

It is, exactly :+1: We don't get the warning for remaining/unhandled `re:run/3` return values anymore as we handle whatever else might be returned explicitly.

In general, I'm trying out approaches to deal with warnings that will reported by Gradualizer. A clause like this seems reasonable as it doesn't require extra options in the typechecker or pragmas in code and is also a reminder to the readers that the catch-all is intentional.

Reply:
It effectively silences the warning, but it goes against good Erlang style by adding defensive code... Well well, it's partly because of re:run/3 returning different stuff depending on options.